native/cocos/math/MathBase.h

```diff
@@ -72,11 +73,14 @@
 NS_CC_MATH_BEGIN
 
 template <typename T, typename Enable = std::enable_if_t<std::is_class<T>::value>>
-struct Hasher final { size_t operator()(const T &info) const; };
+struct Hasher final {
+    size_t operator()(const T &info) const;
```

Comment:
Should return value be hash_t？

Reply:
I think it's probably not, the operator() is used for interacting with `stl containers`.
```c++
ccstd::unordered_map<Vec4, gfx::InputAssembler *, Hasher<Vec4>> _quadIA;
```

So I defined two functions, one is functor for interacting with `stl containers` which return size_t.
Another is a static function for used in our internal code.
```c++
template <typename T, typename Enable = std::enable_if_t<std::is_class<T>::value>>
struct Hasher final {
    size_t operator()(const T &info) const;
    static ccstd::hash_t hashValue(const T &info);
};
```

Using static `hashValue` function could also avoid construct a temporary `Hasher` object.